### PR TITLE
Mbr refactor

### PR DESCRIPTION
**Pull request by Alexander-Sol**
I refactored the MBR function within FlashLFQ. Functions the same, but it's a little bit more modular/easier to work with

### CODE REVIEW

File: mzLib/FlashLFQ/IsotopicEnvelope.cs [hunk 1]
```diff
@@ -18,6 +19,14 @@ public IsotopicEnvelope(IndexedMassSpectralPeak monoisotopicPeak, int chargeStat
             Intensity = intensity / chargeState;
         }
 
+        public IsotopicEnvelope(IndexedMassSpectralPeak monoisotopicPeak, int chargeState, double intensity, double pearsonCorrelation)
```
Comment (nbollis) [inline mzLib/FlashLFQ/IsotopicEnvelope.cs:22]: why do you need a second constructor? If backwards compatability is an issue, could you make pearson correlation have a default in the original constructor?

**Review by nbollis — APPROVED** [review]

File: mzLib/FlashLFQ/FlashLfqEngine.cs [hunk 2]
```diff
@@ -1,16 +1,22 @@
 ﻿using Chemistry;
 using MassSpectrometry;
 using MathNet.Numerics.Distributions;
+using MathNet.Numerics.LinearAlgebra.Factorization;
```
Comment (trishorts) [inline mzLib/FlashLFQ/FlashLfqEngine.cs:4]: some using statements are unused.
image

File: mzLib/FlashLFQ/FlashLfqEngine.cs [hunk 3]
```diff
@@ -469,60 +475,172 @@ private void QuantifyMs2IdentifiedPeptides(SpectraFileInfo fileInfo)
...
 
-        /// <summary>
-        /// This method maps identified peaks from other chromatographic runs ("donors") onto
-        /// the defined chromatographic run ("acceptor"). The goal is to reduce the number of missing
-        /// intensity measurements. Missing values occur generally either because 1) the analyte is
-        /// in the sample but didn't get fragmented/identified or 2) the analyte is genuinely missing
-        /// from the sample.
-        /// </summary>
-        private void QuantifyMatchBetweenRunsPeaks(SpectraFileInfo idAcceptorFile)
+        internal RtInfo PredictRetentionTime(
```
Comment (trishorts) [inline mzLib/FlashLFQ/FlashLfqEngine.cs:478]: please comment this method

File: mzLib/FlashLFQ/FlashLfqEngine.cs [hunk 4]
```diff
@@ -469,60 +475,172 @@ private void QuantifyMs2IdentifiedPeptides(SpectraFileInfo fileInfo)
...
-                .Where(p => p.Condition == idAcceptorFile.Condition && p.BiologicalReplicate == idAcceptorFile.BiologicalReplicate)
-                .Select(p => p.Fraction)
-                .Distinct()
-                .Count() > 1;
 
-            // acceptor file known peaks
-            var acceptorFileIdentifiedPeaks = _results.Peaks[idAcceptorFile];
-            var apexToAcceptorFilePeak = new Dictionary<IndexedMassSpectralPeak, ChromatographicPeak>();
+            var nearbyCalibrationPoints = new List<RetentionTimeCalibDataPoint>();
+            var matchBetweenRunsIdentifiedPeaksThreadSpecific = new Dictionary<string, Dictionary<IsotopicEnvelope, List<ChromatographicPeak>>>();
```
Comment (trishorts) [inline mzLib/FlashLFQ/FlashLfqEngine.cs:486]: unused variable

File: mzLib/FlashLFQ/FlashLfqEngine.cs [hunk 5]
```diff
@@ -469,60 +475,172 @@ private void QuantifyMs2IdentifiedPeptides(SpectraFileInfo fileInfo)
...
+                int acceptorFractionNumber = acceptorFile.Fraction;
+                int donorFractionNumber = donorFile.Fraction;
+
+                if (Math.Abs(acceptorFractionNumber - donorFractionNumber) > 1)
+                {
+                    return null;
+                }
+            }
+
+            Identification donorIdentification = donorPeak.Identifications.OrderBy(p => p.PosteriorErrorProbability).First();
```
Comment (trishorts) [inline mzLib/FlashLFQ/FlashLfqEngine.cs:502]: unused variable

File: mzLib/FlashLFQ/FlashLfqEngine.cs [hunk 6]
```diff
@@ -469,60 +475,172 @@ private void QuantifyMs2IdentifiedPeptides(SpectraFileInfo fileInfo)
...
+                rtInterquartileRange = rtDiffs.InterquartileRange();
+                rtRange = (double)rtInterquartileRange * 4.5; // Multiplication inherited from legacy code, unsure of reason for 4.5
+            }
+
+            rtRange = Math.Min(rtRange, MbrRtWindow);
+
+            return new RtInfo(predictedRt: donorPeak.Apex.IndexedPeak.RetentionTime + median, width: rtRange, rtSd: rtStdDev, rtInterquartileRange: rtInterquartileRange);
+        }
+
+        private MbrScorer BuildMbrScorer(List<ChromatographicPeak> acceptorFileIdentifiedPeaks, out Tolerance fileSpecificMbrTolerance)
```
Comment (trishorts) [inline mzLib/FlashLFQ/FlashLfqEngine.cs:580]: please comment this method

File: mzLib/FlashLFQ/FlashLfqEngine.cs [hunk 7]
```diff
@@ -567,60 +685,18 @@ private void QuantifyMatchBetweenRunsPeaks(SpectraFileInfo idAcceptorFile)
...
 
-                Normal foldChangeDistribution = null;
-                if (listOfFoldChangesBetweenTheFiles.Count > 100)
-                {
-                    new Normal(listOfFoldChangesBetweenTheFiles.Median(), listOfFoldChangesBetweenTheFiles.StandardDeviation());
-                }
+                // generate RT calibration curve
+                RetentionTimeCalibDataPoint[] rtCalibrationCurve = GetRtCalSpline(idDonorFile, idAcceptorFile);
 
+                // Loop through every MSMS id in the donor file
```
Comment (trishorts) [inline mzLib/FlashLFQ/FlashLfqEngine.cs:699]: Line 704 nearbyCalibrationPoints variable never used.

File: mzLib/FlashLFQ/FlashLfqEngine.cs [hunk 8]
```diff
@@ -942,14 +787,161 @@ private void QuantifyMatchBetweenRunsPeaks(SpectraFileInfo idAcceptorFile)
...
 
-                //here is where we get the cosine similarity.
-
                 _results.Peaks[idAcceptorFile].Add(best);
             }
 
             RunErrorChecking(idAcceptorFile);
         }
 
+        internal void FindAllAcceptorPeaks(
```
Comment (trishorts) [inline mzLib/FlashLFQ/FlashLfqEngine.cs:796]: please comment this method

File: mzLib/FlashLFQ/MbrScorer.cs [hunk 9]
```diff
@@ -0,0 +1,140 @@
+﻿using MathNet.Numerics.Distributions;
+using MathNet.Numerics.Statistics;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
```
Comment (trishorts) [inline mzLib/FlashLFQ/MbrScorer.cs:6]: system text and threading are unused

File: mzLib/FlashLFQ/IsotopicEnvelope.cs (hunk 1 above)
Comment (Alexander-Sol) [reply mzLib/FlashLFQ/IsotopicEnvelope.cs]: Good question. Pearson correlation isn't actually used, it was something I had implemented and abandoned. I deleted all references

**Review by nbollis — APPROVED** [review]

**Review by elaboy — APPROVED** [review]

**Review by trishorts — APPROVED** [review]

**Pull request merged**